deltachat-ios/Controller/ChatListController.swift

```diff
@@ -442,8 +442,16 @@ class ChatListController: UITableViewController {
     }
 
     func handleChatListUpdate() {
-        tableView.reloadData()
-        handleEmptyStateLabel()
+        if Thread.isMainThread {
+            tableView.reloadData()
+            handleEmptyStateLabel()
+        } else {
+            DispatchQueue.main.async { [weak self] in
+                guard let self = self else { return }
+                self.tableView.reloadData()
+                self.handleEmptyStateLabel()
+            }
+        }
```

Comment:
in a subsequent pr, we can remove the thread-checks for the other calls of `handleChatListUpdate()`.

Reply:
no, better not, or we would have to add them in the ChatListController used in the Share extension as well.

Reply:
k, then better leave things as is.